pallets/collator-selection/src/lib.rs

```diff
@@ -664,6 +670,88 @@ pub mod pallet {
         }
     }
 
+    /// Checks if a provided NimbusId SessionKey has an associated AccountId
+    impl<T> AccountLookup<T::AccountId> for Pallet<T>
+    where
+        T: pallet_session::Config + Config,
+        // Implemented only where Session's ValidatorId is directly convertible to collator_selection's ValidatorId
+        <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+    {
+        fn lookup_account(author: &NimbusId) -> Option<T::AccountId>
+        where
+            <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+        {
+            use sp_runtime::traits::Convert;
+            #[allow(clippy::bind_instead_of_map)]
+            pallet_session::Pallet::<T>::key_owner(
+                nimbus_primitives::NIMBUS_KEY_ID,
+                &author.to_raw_vec(),
+            )
+            .and_then(|vid| Some(T::AccountIdOf::convert(vid.into())))
+        }
+    }
+
+    /// Fetch list of all possibly eligible authors to use in nimbus consensus filters
+    ///
+    /// This is currently the static set registered with pallet_session and will be superseded by parachain_staking
+    ///
+    /// NOTE: This should really be in pallet_session as we only use its storage, but since we haven't
+    /// forked that one, this is the next best place.
+    impl<T> Get<Vec<T::AccountId>> for Pallet<T>
+    where
+        T: Config + pallet_session::Config,
+        // Implemented only where Session's ValidatorId is directly convertible to collator_selection's ValidatorId
+        <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+    {
+        /// Return the set of eligible collator accounts as registered with pallet session
+        fn get() -> Vec<T::AccountId>
+        where
+            <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+        {
+            use sp_runtime::traits::Convert;
+            pallet_session::Pallet::<T>::validators()
+                .into_iter()
+                .map(
+                    |session_validator_id: <T as pallet_session::Config>::ValidatorId| {
+                        <T as Config>::AccountIdOf::convert(session_validator_id.into())
+                    },
+                )
+                .collect::<Vec<T::AccountId>>()
```

Comment:
Can it be returned as iterator?

Reply:
Unfortunately, the return type is given by the nimbus trait which we don't control :(